src/index.ts

```diff
@@ -67,41 +75,52 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 		return 1;
 	}
 
-	versionA = normalize(versionA);
-	versionB = normalize(versionB);
+	const nVersionA = normalize(versionA);
+	const nVersionB = normalize(versionB);
 
-	const isAValid = semver.valid(versionA);
-	const isBValid = semver.valid(versionB);
+	const isAValid = semver.valid(nVersionA);
+	const isBValid = semver.valid(nVersionB);
 	if (isAValid && !isBValid) {
 		return 1;
 	}
 	if (!isAValid && isBValid) {
 		return -1;
 	}
 	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
+		if (nVersionA > nVersionB) {
 			return 1;
 		}
-		if (versionA < versionB) {
+		if (nVersionA < nVersionB) {
 			return -1;
 		}
 		return 0;
 	}
-	const semverResult = semver.compare(versionA, versionB);
+	const semverResult = semver.compare(nVersionA, nVersionB);
 	if (semverResult !== 0) {
 		return semverResult;
 	}
-	const revA = getRev(versionA);
-	const revB = getRev(versionB);
-	if (revA !== revB) {
-		return revA > revB ? 1 : -1;
-	}
-	const devA = Number(isDevelopmentVersion(versionA));
-	const devB = Number(isDevelopmentVersion(versionB));
-	if (devA !== devB) {
-		return devA > devB ? -1 : 1;
+
+	const suffixComparisonMethods = [
+		{ fn: getRev, reverse: false, normalize: true },
+		{ fn: isPreReleaseVariant, reverse: true, normalize: true },
+		{ fn: getPreReleaseVariant, reverse: false, normalize: true },
```

Comment:
> Obviously, if we are dropping -dev/-prod prereleases, then we can get rid of it.

It's not that we shouldn't care about `-dev` or `-prod` at all, it's just that they're always _part_ of a prerelease version, but not necessarily at the front (and so far, _never_ at the front). The real versions so far look like `1.2.3-beta.dev`, so `-dev` is unlikely to ever match.

Anyway, I've just done some investigation, and in fact this comparison isn't doing anything at all. `semver` already compares prerelease versions (part by part, alphabetically: https://github.com/npm/node-semver/blob/master/semver.js#L356-L384), so at this point in the code we _know_ that the two versions already have exactly the same prerelease version (potentially none for both).

That means we can actually drop both prelease tests here completely - they never make a difference.